### ISSUE TITLE
An exception occurred after updating the Jackson library(2.13.2.2) (dependency error?)

### ISSUE THREAD

**Issue by ALtKEy** [Azure/azure-sdk-for-java#27990]
**Describe the bug**
An exception occurred after updating the Jackson library

[FasterXML Jackson CVE library update](https://github.com/FasterXML/jackson-databind/issues/2816)

***Exception or Stack Trace***
> 2022-04-01T11:02:14,054 INFO  [https-jsse-nio-18443-exec-63] [c.a.c.i.j.JacksonVersion:222] - Package versions: jackson-annotations=2.13.2, jackson-core=2.13.2, jackson-databind=2.13.2.2, jackson-dataformat-xml=2.13.2, jackson-datatype-jsr310=2.13.2, azure-core=1.25.0, Troubleshooting version conflicts: https://aka.ms/azsdk/java/dependency/troubleshoot
2022-04-01T11:02:14,056 DEBUG [https-jsse-nio-18443-exec-63] [c.a.c.i.j.ObjectMapperFactory:491] - Failed to retrieve MethodHandles used to set coercion configurations. Setting coercion configurations will be skipped.
java.lang.IllegalAccessException: no such method: com.fasterxml.jackson.databind.ObjectMapper.coercionConfigDefaults()MutableCoercionConfig/invokeVirtual
        at java.lang.invoke.MemberName.makeAccessException(MemberName.java:867) ~[?:1.8.0_312]
        at java.lang.invoke.MemberName$Factory.resolveOrFail(MemberName.java:1003) ~[?:1.8.0_312]
        at java.lang.invoke.MethodHandles$Lookup.resolveOrFail(MethodHandles.java:1386) ~[?:1.8.0_312]
        at java.lang.invoke.MethodHandles$Lookup.findVirtual(MethodHandles.java:861) ~[?:1.8.0_312]
        at com.azure.core.implementation.jackson.ObjectMapperFactory.<init>(ObjectMapperFactory.java:53) ~[azure-core-1.25.0.jar:1.25.0]
        at com.azure.core.implementation.jackson.ObjectMapperFactory.<clinit>(ObjectMapperFactory.java:43) ~[azure-core-1.25.0.jar:1.25.0]
        at com.azure.core.implementation.jackson.ObjectMapperShim.createHeaderMapper(ObjectMapperShim.java:126) ~[azure-core-1.25.0.jar:1.25.0]
        at com.azure.core.util.serializer.JacksonAdapter.<init>(JacksonAdapter.java:80) ~[azure-core-1.25.0.jar:1.25.0]
        at com.azure.core.util.serializer.JacksonAdapter.<init>(JacksonAdapter.java:58) ~[azure-core-1.25.0.jar:1.25.0]
        at com.azure.core.util.serializer.JacksonAdapter$SerializerAdapterHolder.<clinit>(JacksonAdapter.java:113) ~[azure-core-1.25.0.jar:1.25.0]
        at com.azure.core.util.serializer.JacksonAdapter.createDefaultSerializerAdapter(JacksonAdapter.java:122) ~[azure-core-1.25.0.jar:1.25.0]
        at com.azure.identity.implementation.IdentityClient.<clinit>(IdentityClient.java:101) ~[azure-identity-1.4.6.jar:1.4.6]
        at com.azure.identity.implementation.IdentityClientBuilder.build(IdentityClientBuilder.java:139) ~[azure-identity-1.4.6.jar:1.4.6]
        at com.azure.identity.ClientSecretCredential.<init>(ClientSecretCredential.java:68) ~[azure-identity-1.4.6.jar:1.4.6]
        at com.azure.identity.ClientSecretCredentialBuilder.build(ClientSecretCredentialBuilder.java:93) ~[azure-identity-1.4.6.jar:1.4.6]

**Setup (please complete the following information):**
 - OS: Windows 10 (21H2 19044.1586)
 - IDE: Eclipse IDE (2022.03)
 - Library/Libraries: azure-core(1.26.0), azure-identity(1.4.6), msal4j(1.11.3), microsoft-graph(5.18.0), Spring framework BOM(5.3.17), Jackson BOM (2.13.2.20220328)
 - Java version: Eclipse Adoptium Temurin OpenJDK 1.8.322.6-hotspot
 - App Server/Environment: Apache Tomcat 9.0.59
 - Frameworks: Spring framework (/with web-mvc)

**Using Jar full list**
accessors-smart-2.4.7.jar
annotations-13.0.jar
asm-9.1.jar
azure-core-1.25.0.jar
azure-core-http-netty-1.11.8.jar
azure-identity-1.4.6.jar
checker-qual-3.12.0.jar
commons-codec-1.15.jar
commons-io-2.11.0.jar
commons-lang3-3.12.0.jar
content-type-2.2.jar
error_prone_annotations-2.11.0.jar
failureaccess-1.0.1.jar
gson-2.9.0.jar
guava-31.1-jre.jar
j2objc-annotations-1.3.jar
jackson-annotations-2.13.2.jar
jackson-core-2.13.2.jar
jackson-databind-2.13.2.2.jar
jackson-dataformat-xml-2.13.2.jar
jackson-datatype-jsr310-2.13.2.jar
jcip-annotations-1.0-1.jar
jna-5.5.0.jar
jna-platform-5.6.0.jar
json-smart-2.4.7.jar
jsr305-3.0.2.jar
kotlin-stdlib-1.4.10.jar
kotlin-stdlib-common-1.4.0.jar
lang-tag-1.5.jar
listenablefuture-9999.0-empty-to-avoid-conflict-with-guava.jar
log4j-api-2.17.2.jar
log4j-core-2.17.2.jar
log4j-over-slf4j-1.7.36.jar
log4j-slf4j-impl-2.17.2.jar
microsoft-graph-5.18.0.jar
microsoft-graph-core-2.0.11.jar
msal4j-1.11.3.jar
msal4j-persistence-extension-1.1.0.jar
netty-buffer-4.1.73.Final.jar
netty-codec-4.1.73.Final.jar
netty-codec-dns-4.1.72.Final.jar
netty-codec-http-4.1.73.Final.jar
netty-codec-http2-4.1.73.Final.jar
netty-codec-socks-4.1.73.Final.jar
netty-common-4.1.73.Final.jar
netty-handler-4.1.73.Final.jar
netty-handler-proxy-4.1.73.Final.jar
netty-resolver-4.1.73.Final.jar
netty-resolver-dns-4.1.72.Final.jar
netty-resolver-dns-classes-macos-4.1.72.Final.jar
netty-resolver-dns-native-macos-4.1.72.Final-osx-x86_64.jar
netty-tcnative-boringssl-static-2.0.46.Final.jar
netty-tcnative-classes-2.0.46.Final.jar
netty-transport-4.1.73.Final.jar
netty-transport-classes-epoll-4.1.73.Final.jar
netty-transport-classes-kqueue-4.1.73.Final.jar
netty-transport-native-epoll-4.1.73.Final-linux-x86_64.jar
netty-transport-native-kqueue-4.1.73.Final-osx-x86_64.jar
netty-transport-native-unix-common-4.1.73.Final.jar
nimbus-jose-jwt-9.15.2.jar
oauth2-oidc-sdk-9.22.1.jar
okhttp-4.9.3.jar
okio-2.8.0.jar
reactive-streams-1.0.3.jar
reactor-core-3.4.13.jar
reactor-netty-core-1.0.15.jar
reactor-netty-http-1.0.15.jar
slf4j-api-1.7.36.jar
spring-aop-5.3.17.jar
spring-beans-5.3.17.jar
spring-context-5.3.17.jar
spring-core-5.3.17.jar
spring-expression-5.3.17.jar
spring-jcl-5.3.17.jar
spring-web-5.3.17.jar
spring-webmvc-5.3.17.jar
stax2-api-4.2.1.jar
woodstox-core-6.2.7.jar

**Comment by ALtKEy** [comment]
[maven_dependency_tree.txt](https://github.com/Azure/azure-sdk-for-java/files/8394238/maven_dependency_tree.txt)

**Comment by joshfree** [comment]
@alzimmermsft could you please follow up with @ALtKEy?

**Comment by alzimmermsft** [comment]
Thanks for filing this issue @ALtKEy, I wasn't able to replicate this without using Tomcat (in the hope the issue was more obvious). In your Tomcat deployment, is there any special handling with classloaders, or if using an uber JAR is there anything being excluded? The APIs that azure-core is failing to find continue existing between the patch/point patch versions that azure-core 1.26.0 and the jackson-bom you're using.

**Comment by ALtKEy** [comment]
@alzimmermsft Thank you for your reply. 
1. In your Tomcat deployment, is there any special handling with classloaders
    - No special handling classloaders
2. no uber JAR file included

As an additional confirmation, an error appears only on one call, and no error occurs on subsequent re-requests.
As you know, restarting WAS, an error appears only on one call, and no error occurs on subsequent re-requests.

-- More log

"
Caused by: java.lang.LinkageError: loader constraint violation: when resolving method "com.fasterxml.jackson.databind.ObjectMapper.coercionConfigDefaults()Lcom/fasterxml/jackson/databind/cfg/MutableCoercionConfig;" the class loader (instance of <bootloader>) of the current class, java/lang/Object, and the class loader (instance of org/apache/catalina/loader/ParallelWebappClassLoader) for the method's defining class, com/fasterxml/jackson/databind/ObjectMapper, have different Class objects for the type com/fasterxml/jackson/databind/cfg/MutableCoercionConfig used in the signature
        at java.lang.invoke.MethodHandleNatives.resolve(Native Method) ~[?:1.8.0_312]
        at java.lang.invoke.MemberName$Factory.resolve(MemberName.java:975) ~[?:1.8.0_312]
        at java.lang.invoke.MemberName$Factory.resolveOrFail(MemberName.java:1000) ~[?:1.8.0_312]
"

2022-04-05T11:33:03,735 DEBUG [https-jsse-nio-18443-exec-146] [c.a.c.i.j.ObjectMapperFactory:174] - Didn't set coercion defaults as it wasn't found on the classpath.
2022-04-05T11:33:03,742 DEBUG [https-jsse-nio-18443-exec-146] [r.u.Loggers:245] - Using Slf4j logging framework

**Comment by alzimmermsft** [comment]
Hi @ALtKEy, I created a simple WAR in attempt to replicate this error using the version of Tomcat you've included but I wasn't able to have the error reproduce. Attached is the simple application I created which includes a call using the method that threw the exception in your stack trace. I used the basic Tomcat setup based on their documentation where I installed Tomcat and set CATILINA_HOME, then used the web GUI to deploy the WAR generated by the project I included. When I navigated to the page that triggers `doGet` the creation of the `ClientSecretCredential` doesn't throw the exception as I was hoping it would.

Is the Tomcat setup you're using more complex, such as shared class loaders, or is the application you're using more complex? Also, is it using a WAR or the source directory?

[Issue27990.zip](https://github.com/Azure/azure-sdk-for-java/files/8445347/Issue27990.zip)
